app/lib/account/session_cookie.dart

```diff
@@ -89,6 +92,36 @@ String? parseUserSessionCookie(String? cookieString) {
   return sessionId;
 }
 
+/// Parses the cookie values and returns the status of client session cookies.
+ClientSessionCookieStatus parseClientSessionCookies(String? cookieString) {
```

Comment:
If instead we did:
`parseClientSessionCookies(String? cookieString, String method)`

We can just return `sessionId`.

Even better, we could do: `parseClientSessionCookies(shelf.Request request)`

In cases where we have inconsistency can consider throwing a RequestException that clears the cookies?

Then we wouldn't have to return `needsReset`.

Reply:
I think it is better to keep this method at the cookie-parsing logic, and have another one which authorizes the cookie based on the method and the attempted operation.